syntax = "proto3";                                                     

option java_multiple_files = true;
option java_package = "com.xiaomi.ai.test";                   
option java_outer_classname = "HelloWorldProto";                       
option objc_class_prefix = "HLW";                                      

package im2txt;                                                    

// The greeting service definition.                                    
service ShowAndTellService {
  // Sends a greeting
  rpc ShowAndTell(ShowAndTellRequest) returns (ShowAndTellReply) {}                  
}                                                                      

// The request message containing the user's name.                     
message ShowAndTellRequest {                                                 
  bytes image_data = 1;                                                     
}                                                                      

// The response message containing the greetings                       
message ShowAndTellReply {
  message Caption {
    string caption = 1;
    float score = 2;
  }
  repeated Caption captions = 1;                                                  
}

